File: OnlinePK-Android/app/src/main/java/com/netease/yunxin/nertc/demo/list/FunctionItem.java
/*
 * // Copyright (c) 2021 NetEase, Inc.  All rights reserved.
 * // Use of this source code is governed by a MIT license that can be found in the LICENSE file.
 */

package com.netease.yunxin.nertc.demo.list;

public class FunctionItem {
    public int type = FunctionAdapter.TYPE_VIEW_CONTENT;

    public final int iconResId;

    public final String nameStr;

    public final String descriptionStr;

    public final Runnable action;

    public FunctionItem(int type, String nameStr,int iconResId) {
        this.type = type;
        this.nameStr = nameStr;
        this.descriptionStr = null;
        this.iconResId = iconResId;
        this.action = null;
    }

    public FunctionItem(int iconResId, String nameStr,String descriptionStr, Runnable action) {
        this.iconResId = iconResId;
        this.nameStr = nameStr;
        this.descriptionStr = descriptionStr;
        this.action = action;
    }
}

File: OnlinePK-Android/app/src/main/java/com/netease/yunxin/nertc/demo/list/FunctionItem.kt
/*
 *  Copyright (c) 2021 NetEase, Inc.  All rights reserved.
 *  Use of this source code is governed by a MIT license that can be found in the LICENSE file
 */

package com.netease.yunxin.nertc.demo.list

class FunctionItem {
    var type: Int = FunctionAdapter.Companion.TYPE_VIEW_CONTENT
    val iconResId: Int
    val nameStr: String
    val descriptionStr: String?
    val action: Runnable?

    constructor(type: Int, nameStr: String, iconResId: Int) {
        this.type = type
        this.nameStr = nameStr
        descriptionStr = null
        this.iconResId = iconResId
        action = null
    }

    constructor(iconResId: Int, nameStr: String, descriptionStr: String?, action: Runnable?) {
        this.iconResId = iconResId
        this.nameStr = nameStr
        this.descriptionStr = descriptionStr
        this.action = action
    }
}